ansible/roles/accumulo/tasks/download.yml

```diff
@@ -19,10 +19,10 @@
   shell: curl -sk https://apache.org/mirrors.cgi?as_json | grep preferred | cut -d \" -f 4
   args:
     warn: no
+  register: apache_mirror
   retries: 10
   delay: 10
-  register: apache_mirror
-  failed_when: "'http' not in apache_mirror.stdout"
+  until: "'http' in apache_mirror.stdout"
```

Comment:
The retrieval of the mirror list is a bit unreliable, since it is depending on the unstable json formatting that the server just happens to return. If we could ensure the machines had `jq` installed, we could easily parse the json instead. Something like:

```yaml
shell: curl -sk https://apache.org/mirrors.cgi?as_json | jq -r .preferred
```

I'm not sure if it's possible to base the retry on the the command return codes, but if it is, you AND the return codes from both commands in the pipe with something like:

```yaml
shell: curl -sk https://apache.org/mirrors.cgi?as_json | jq -r .preferred; [[ ${PIPESTATUS[0]} -eq 0 && ${PIPESTATUS[1]}
```

I don't know YAML/Ansible well enough to know how much bash could work there without getting crazy with the escaping, but if you could retry until the command is successful, that might be better than checking for 'http' in the output... which could also appear in an error message (and also 'preferred' could return ftp).

At the very least, parsing using `jq` instead of `grep` could improve this retrieval / retry code a bit. (Or, if you don't want to install `jq`, you could do the same in python, but it's uglier: `| python3 -c "import sys, json; print(json.load(sys.stdin)['preferred'])"`).

Reply:
Thank you for your inputs, @ctubbsii ! Based on the same I have incorporated jq to parse the JSON and also cleaned up the code to reference a shared task. I also tested and verified that things are work correctly in both happy path and failure path (i.e. the limited retries mechanism are working correctly).